include/yarp/util/yp_newline_list.h

```diff
@@ -0,0 +1,52 @@
+// When compiling the syntax tree, it's necessary to know the line and column
+// of many nodes. This is necessary to support things like error messages,
+// tracepoints, etc.
+//
+// It's possible that we could store the start line, start column, end line, and
+// end column on every node in addition to the offsets that we already store,
+// but that would be quite a lot of memory overhead.
+
+#ifndef YP_NEWLINE_LIST_H
+#define YP_NEWLINE_LIST_H
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+// A list of offsets of newlines in a string. The offsets are assumed to be
+// sorted/inserted in ascending order.
+typedef struct {
+    const char *start;
+
+    size_t *offsets;
```

Comment:
Are we going to try trend away from `size_t`s?

Reply:
I think this is fine because it won't be serialized. When moving away from `size_t` I was mostly talking about it in the context of having to change them all to `uint32_t` for explicit widths in the serialization.